e2e/test/iothub/device/FileUploadE2ETests.cs

```diff
@@ -142,13 +142,24 @@ await using (deviceClient)
                 {
                     var notification = new FileUploadCompletionNotification("invalid-correlation-id", uploadTask.IsCompleted);
 
-                    // act
-                    Func<Task> act = async () => await deviceClient.CompleteFileUploadAsync(notification).ConfigureAwait(false);
-
-                    // assert
-                    var error = await act.Should().ThrowAsync<IotHubClientException>();
-                    error.And.ErrorCode.Should().Be(IotHubClientErrorCode.ServerError);
-                    error.And.IsTransient.Should().BeTrue();
+                    // act and assert
+
+                    try
+                    {
+                        // TODO: the HTTP layer handles errors differently and does not produce the right kind of exceptions.
```

Comment:
@brycewang-microsoft you might be interested in taking on this work. In Gateway V2 we now get a 400-level HTTP status code, which translates to `ArgumentException` instead of `IotHubClientException` with a relevant error code.

I'll add a bug or PBI for it.